blockchain/blockchain.go

```diff
@@ -1588,7 +1588,15 @@ func (bc *BlockChain) insertChain(chain types.Blocks) (int, []interface{}, []*ty
 		seals[i] = true
 	}
 
-	abort, results := bc.engine.VerifyHeaders(bc, headers, seals)
+	var (
+		abort   chan<- struct{}
+		results <-chan error
+	)
+	if bc.engine.CanVerifyHeadersConcurrently() {
+		abort, results = bc.engine.VerifyHeaders(bc, headers, seals)
+	} else {
+		abort, results = bc.engine.PreprocessHeaderVerification(headers)
+	}
```

Comment:
`PreprocessHeaderVerification` executes some parts of `VerifyHeaders` and the rest parts of the verfication are not executed if `bc.engine.CanVerifyHeadersConcurrently()` is false.

Reply:
Right. `VerifyHeaders` is enough to implement the functionality, but the new method is introduced since the `PreprocessHeaderVerification` does not verify headers. Any suggestion?